pallets/order-book/src/lib.rs

```diff
@@ -644,37 +646,100 @@ impl<T: Config> LiquiditySource<T::DEXId, T::AccountId, T::AssetId, Balance, Dis
         let deal_info =
             order_book.calculate_deal(input_asset_id, output_asset_id, amount, &mut data)?;
 
+        ensure!(deal_info.is_valid(), Error::<T>::PriceCalculationFailed);
+
         let fee = 0; // todo (m.tagirov)
 
         match amount {
             QuoteAmount::WithDesiredInput { .. } => Ok((
-                SwapOutcome::new(deal_info.output_amount, fee),
+                SwapOutcome::new(*deal_info.output_amount.value(), fee),
                 Self::quote_weight(),
             )),
             QuoteAmount::WithDesiredOutput { .. } => Ok((
-                SwapOutcome::new(deal_info.input_amount, fee),
+                SwapOutcome::new(*deal_info.input_amount.value(), fee),
                 Self::quote_weight(),
             )),
         }
     }
 
     fn exchange(
```

Comment:
Just to clarify: the code that will call this function will be added later? Something like calling from liquidity proxy, right?

Reply:
It is already added in #444 
It is called from `liquidity-proxy`

Reply:
Oh, ok. Couldn't find it at first because it uses macro `exchange!` instead of usual function call.
Thanks, I see it now